server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -420,6 +513,10 @@ void setPrefixFieldType(PrefixFieldType prefixFieldType) {
             this.prefixFieldType = prefixFieldType;
         }
 
+        void indexPhrases(boolean indexPhrases) {
+            this.indexPhrases = indexPhrases;
+        }
```

Comment:
you should update `TextFieldTypeTests` to add a new modifier for this property

Reply:
I'd missed TextFieldTypeTest - I also need to add in checks for `index_prefix` here.  I agree with you that it's odd to have `index_prefix` and `index_phrases`, so I think we should change the former to be plural as well.  I'll open a separate PR for that change, and to add the TextFieldTypeTest modifications for `index_prefix`